ql/src/test/org/apache/hadoop/hive/ql/udf/generic/TestGenericUDFJsonRead.java

```diff
@@ -156,10 +158,8 @@ public void testUndeclaredStructField() throws Exception {
       ObjectInspector[] arguments = buildArguments("struct<a:int>");
       udf.initialize(arguments);
 
-      Object res = udf.evaluate(evalArgs("{\"b\":null}"));
-      assertTrue(res instanceof Object[]);
-      Object o[] = (Object[]) res;
-      assertEquals(null, o[0]);
+      // Invalid - should throw Exception
+      udf.evaluate(evalArgs("{\"b\":null}"));
```

Comment:
am Not sure why this has changed ? seems like this is different from old behavior can you please explain more?

Reply:
Good question.

If you checkout the test method, it is annotated with `@Test(expected = HiveException.class)`.   This means that the test will fail if it does not throw a HiveException.   What causes it to throw the Exception is the call to `udf.evaluate(evalArgs("{\"b\":null}"))`  therefore, everything that comes after it is dead code, so I simply removed it.